### DISCUSSION TITLE
'packageRules > schedule' isn't behaving as expected part II

### DISCUSSION THREAD

**Discussion by kipusoep** [renovatebot/renovate#25829] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.60.2

### If you're self-hosting Renovate, select which platform you are using.

Azure DevOps (dev.azure.com)

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Given the following config:

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "enabledManagers": [ "nuget", "dockerfile" ],
  "rebaseWhen": "behind-base-branch",
  "labels": [ "Renovate", "BackEnd" ],
  "internalChecksFilter": "strict",
  "major": {
    "enabled": false
  },
  "groupName": "all packages",
  "prHourlyLimit": 0,
  "packageRules": [
    {
      "matchUpdateTypes": [ "minor", "patch", "pin", "digest" ], // All 3rd party packages
      "stabilityDays": 7,
      "extends": [ "schedule:weekends" ]
    },
    {
      "matchPackagePrefixes": [ "Contoso." ], // All internal company packages
      "stabilityDays": 0,
      "schedule": [ "at any time" ]
    }
  ]
}
```

We run the renovate pipeline scheduled on Saturday, to update 3rd party packages and internal company packages. They are combined in a single PR thanks to the groupName 'all packages'.
But we also run renovate based on a trigger when one of our internal company packages updates.
As the schedule is 'at any time', I would expect a PR to be created.
But the logs indicate that the weekends schedule is used instead:

```
DEBUG: Checking schedule(every weekend, null) (repository=Contoso/RenovateTests, branch=renovate/all-packages)

DEBUG: Checking 1 schedule(s) (repository=Contoso/RenovateTests, branch=renovate/all-packages)

DEBUG: Checking schedule "every weekend" (repository=Contoso/RenovateTests, branch=renovate/all-packages)
       "parsedSchedule": {"schedules": [{"d": [1, 7]}], "exceptions": [], "error": -1}

DEBUG: Package not scheduled (repository=Contoso/RenovateTests, branch=renovate/all-packages)

DEBUG: Skipping branch creation as not within schedule (repository=Contoso/RenovateTests, branch=renovate/all-packages)
```

I would expect evaluation of both schedules.

### What you tried so far.

See above. I guess it's a bit related to #7952?

### Relevant debug logs

See above.

**Comment by rarkins** [comment]
I don't think it makes sense for a branch to have multiple schedules, and it certainly can't guess that you're running it because of an internal company package update.

If you wish for certain packages to have different schedules to their default, you need them to be in their own branch/PR with a separate schedule.

**Reply by kipusoep** [reply]
That's a shame. The thing is, we're using a microservices architecture and it's a lot of work to run by all the renovate PRs on Monday.
Besides that, having multiple PRs per repository is giving us conflict issues in the CI pipeline. That's why we'd like to group all updates togethes in a single PR.

So bottom line, it's not possible at all to configure it the way we would like it to? Also not using a work-around, like having 2 seperate renovate.json files; 1 for the weekend and 1 for the weekdays?

**Reply by rarkins** [reply]
It's not possible to have two schedules per branch, assuming that's what you mean by "like it to".

You also cannot have two configs per repo, but you can naturally change the bot's config every time you run it, and could configure a rule to break your internal package(s) out of the group at specific times. You might make use of `prPriority` for that purpose

**Reply by kipusoep** [reply]
I tried some things, but can't get it to work.
If I understand you correctly, you're suggesting to use the CLI arguments to conditionally change the config, based on whether or not it's weekend, right?
I tried adding a package rule using the CLI arg, but didn't seem to work. Is there an example how to format this?
Next idea I had was to set the --group-name, but that also doesn't seem to work. I can see the correct value in the logs, but are the CLI args merged into the final config with the renovate.json config?

I appreciate the effort by the way :-)

**Reply by rarkins** [reply]
I would suggest using two different config files, or a single config file (.js) which switches config based on an environment variable. You can choose which config file to use using `RENOVATE_CONFIG_FILE`
